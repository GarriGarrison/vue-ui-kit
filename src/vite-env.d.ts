/* eslint-disable @typescript-eslint/no-explicit-any */
/// <reference types="vite/client" />

//* fixing fetch
interface Body {
  json(): Promise<unknown>;
}

//* fixing JSON.parse
interface JSON {
  parse(text: string, reviver?: (tis: any, key: string, value: any) => any): unknown;
}

//* fixing structuredClone
declare function structuredClone<T>(value: T, options?: StructuredSerializeOptions): T;

//* fixing Array.isArray
interface ArrayConstructor {
  isArray(arg: any): arg is unknown[];
}
