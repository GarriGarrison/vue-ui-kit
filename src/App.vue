<script setup lang="ts">
import { GuvButton } from '.';
</script>

<template>
  <guv-button>Click me</guv-button>
</template>
