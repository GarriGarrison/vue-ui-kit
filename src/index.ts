export { default as GuvBadge } from './components/badge/guv-badge.vue';
export { default as GuvButton } from './components/button/guv-button.vue';
export { default as GuvInput } from './components/input/guv-input.vue';
export { default as GuvView } from './components/view/guv-view.vue';
