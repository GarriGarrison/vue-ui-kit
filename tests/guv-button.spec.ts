import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import { GuvButton as Button } from '../src/';

describe('Button', () => {
  /**
   * Positive testing
   */
  it('Title -> test', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      props: {
        title: 'Click me',
      },
    });
    expect(wrapper.html()).toContain('Click me');
  });

  it('Title -> undefined', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button);
    expect(wrapper.html()).toContain('');
  });

  it('Slots', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      slots: {
        before: 'Before',
        after: '<div>After</div>',
      },
    });
    expect(wrapper.html()).toContain('Before');
    expect(wrapper.html()).toContain('<div>After</div>');
  });

  it('Size -> standard', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      props: {
        size: 'standard',
      },
    });
    expect(wrapper.classes()).toContain('standard');
  });

  it('Size -> small', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      props: {
        size: 'small',
      },
    });
    expect(wrapper.classes()).toContain('small');
  });

  it('Classes -> main', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button);
    expect(wrapper.classes()).toContain('guv-button');
  });

  it('Classes -> test', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      props: {
        class: 'test',
      },
    });
    expect(wrapper.classes()).toContain('test');
  });

  it('Style -> add', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      props: {
        style: { paddingTop: '10px' },
      },
    });
    expect(wrapper.attributes().style).toContain('padding-top: 10px;');
  });

  it('Emit -> mounted', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button);
    expect(wrapper.emitted()).toHaveProperty('mount');
  });

  it('Emit -> unmounted', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button);
    wrapper.unmount();
    expect(wrapper.emitted()).toHaveProperty('unmount');
  });

  it('Emit -> click', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button);
    wrapper.trigger('click');
    expect(wrapper.emitted()).toHaveProperty('click');
  });

  it('Disabled', () => {
    expect(Button).toBeTruthy();

    const wrapper = mount(Button, {
      props: {
        disabled: true,
      },
    });

    expect(wrapper.classes()).toContain('disabled');

    wrapper.trigger('click');
    expect(wrapper.emitted()).not.toHaveProperty('click');
  });
});
