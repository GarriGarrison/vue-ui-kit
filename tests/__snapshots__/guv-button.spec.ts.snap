// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > Shapshot 1`] = `
<button
  class="guv-button"
  data-v-43f84a0c=""
>
  
  
  
  Button
  
  
  
</button>
`;
